tools/build-version.go

```diff
@@ -42,6 +42,10 @@ func main() {
 	version, err := semver.ParseTolerant(versionStr)
 	if err != nil {
 		// no version tag found so just return what ever we can find.
+		if hash, err := exec.Command("git", "rev-parse", "--short", "HEAD").Output(); err == nil {
```

Comment:
Print the error if it failed?
Hmm... errors are not printed in the rest of this code either...

Reply:
It shall print the hash in case there is no error and in case it is, then it shall proceed with the already known `0.0.0-unknown`:
https://github.com/zyedidia/micro/actions/runs/9534884176/job/26279918356?pr=3354#step:4:8

Reply:
Yeah, but my point is that ideally we probably also want to print the error, with something like 

```
log.Print("git rev-parse: ", err)
if e, ok := err.(*exec.ExitError); ok {
	log.Print(string(e.Stderr))
}
```

so that we can find out why exactly it failed and we had to fallback to `0.0.0-unknown`.

...More importantly: do we really understand why was the tag not found (i.e. why is this PR needed at all)? We have this:

```
	if tags, err := exec.Command("git", "tag").Output(); err != nil || len(tags) == 0 {
		// no tags found -- fetch them
		exec.Command("git", "fetch", "--tags").Run()
	}
```

so the tag should have been found, right?

When I run `go run tools/build-version.go` in my micro tree, on the master branch, I get:

```
2.0.14-dev.207
```

That is probably what we should already get, instead of `0.0.0-unknown`. Why don't we? Does it fail to run `exec.Command("git", "fetch", "--tags").Run()`? Or something else failed? We should add error printing to all those places, to find out what's going on, right?

Reply:
You're right, I've overseen the line...
`exec.Command("git", "fetch", "--tags").Run()`
...which obviously should get the tags, but didn't (even in the [test.yaml](https://github.com/zyedidia/micro/actions/workflows/test.yaml)). Locally everything is fine...

Then I will update the PR to make these helper scripts more error verbose. To not workaround the problem I will remove the `fetch-tags` from the checkout script.

Reply:
Damn, https://stackoverflow.com/questions/4916492/git-describe-fails-with-fatal-no-names-found-cannot-describe-anything brought the necessary hint to bypass that stuff.
Even `fetch-tags: true` didn't help.